test/spec/modules/jcmBidAdapter_spec.js

```diff
@@ -1,244 +1,123 @@
-    }
-    if (typeof ($$PREBID_GLOBAL$$._adsReceived) === 'undefined') {
-      $$PREBID_GLOBAL$$._adsReceived = [];
-    }
-
-    it('should be called only once', function () {
-      var params = {
-        bidderCode: 'jcm',
-        bidder: 'jcm',
-        bids: [
-          {
-		          bidId: '3c9408cdbf2f68',
-     		          sizes: [[300, 250], [300, 300]],
-            bidder: 'jcm',
-            params: { siteId: '3608', adSizes: '300x250' },
-            requestId: '10b327aa396609',
-            placementCode: '/19968336/header-bid-tag-0'
-          }
-
-        ]
-      };
-
-      adapter().callBids(params);
-
-      sinon.assert.calledOnce(stubLoadScript);
+    it('should return true when required params found', () => {
+      expect(spec.isBidRequestValid(bid)).to.equal(true);
     });
 
-    it('should fix parameter name', function () {
-      var params = {
-        bidderCode: 'jcm',
-        bidder: 'jcm',
-        bids: [
-          {
-            bidId: '3c9408cdbf2f68',
-            sizes: [[300, 250]],
-            bidder: 'jcm',
-            params: { siteId: '3608', adSizes: '300x250' },
-            requestId: '10b327aa396609',
-            placementCode: '/19968336/header-bid-tag-0'
-          }
-
-        ]
-      };
-
-      adapter().callBids(params);
-      var bidUrl = stubLoadScript.getCall(0).args[0];
-
-      sinon.assert.calledWith(stubLoadScript, bidUrl);
-
-      var parsedBidUrl = urlParse(bidUrl);
-      var parsedBidUrlQueryString = querystringify.parse(parsedBidUrl.query);
-
-      expect(parsedBidUrl.hostname).to.equal('media.adfrontiers.com');
-      expect(parsedBidUrl.pathname).to.equal('/pq');
+    it('should return false when required params are not passed', () => {
+      let bid = Object.assign({}, bid);
+      delete bid.params;
+      expect(spec.isBidRequestValid(bid)).to.equal(false);
+    });
+  });
 
-      expect(parsedBidUrlQueryString).to.have.property('t').and.to.equal('hb');
-      expect(parsedBidUrlQueryString).to.have.property('bids');
+  describe('buildRequests', () => {
+    let bidRequests = [
+      {
+        'bidder': 'jcm',
+        'params': {
+          'siteId': '3608'
+        },
+        'adUnitCode': 'adunit-code',
+        'sizes': [[300, 250], [300, 600]],
+        'bidId': '30b31c1838de1e',
+        'bidderRequestId': '22edbae2733bf6',
+        'auctionId': '1d1a030790a475',
+      }
 
-      var bidObjArr = JSON.parse(parsedBidUrlQueryString.bids);
-      expect(bidObjArr).to.have.property('bids');
-      var bidObj = bidObjArr.bids[0];
+    ];
 
-      expect(bidObj).to.have.property('adSizes').and.to.equal('300x250');
-      expect(bidObj).to.have.property('siteId').and.to.equal('3608');
-      expect(bidObj).to.have.property('callbackId').and.to.equal('3c9408cdbf2f68');
+    it('sends bid request to ENDPOINT via GET', () => {
+      const request = spec.buildRequests(bidRequests);
+      expect(request.method).to.equal('GET');
```

Comment:
Needs some real assertions... barely tests `buildRequests` at all.

Reply:
Will look into this.